consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -157,6 +181,6 @@ public ConsulUpdatingListenerBuilder consulToken(String consulToken) {
      */
     public ConsulUpdatingListener build() {
         return new ConsulUpdatingListener(consulClientBuilder.build(), serviceName, serviceEndpoint,
-                                          checkUri, checkMethod, checkInterval);
+                                          checkUri, checkMethod, checkInterval, tags);
```

Comment:
I think you can keep the tag as an immutable list internally so that it doesn't copy to the 'Service' object.
`tagsBuilder.build().asList()`